runtime/signal.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+#[cfg(target_os = "windows")]
+#[derive(Debug, thiserror::Error)]
+#[error("Windows only supports ctrl-c (SIGINT) and ctrl-break (SIGBREAK), but got {0}")]
+pub struct InvalidSignalStrError(pub String);
+
+#[cfg(any(
+  target_os = "android",
+  target_os = "linux",
+  target_os = "openbsd",
+  target_os = "openbsd",
+  target_os = "macos",
+  target_os = "solaris",
+  target_os = "illumos"
+))]
+#[derive(Debug, thiserror::Error)]
+#[error("Invalid signal: {0}")]
+pub struct InvalidSignalStrError(pub String);
+
+#[cfg(target_os = "windows")]
+#[derive(Debug, thiserror::Error)]
+#[error("Windows only supports ctrl-c (SIGINT) and ctrl-break (SIGBREAK), but got {0}")]
+pub struct InvalidSignalIntError(pub libc::c_int);
+
+#[cfg(any(
+  target_os = "android",
+  target_os = "linux",
+  target_os = "openbsd",
+  target_os = "openbsd",
+  target_os = "macos",
+  target_os = "solaris",
+  target_os = "illumos"
+))]
+#[derive(Debug, thiserror::Error)]
+#[error("Invalid signal: {0}")]
+pub struct InvalidSignalIntError(pub libc::c_int);
```

Comment:
Everything is the same in this file except these new errors and it adds a `SIGNAL_NUMS` array for getting a list of signals that the operating system can listen to.